content/recipes/mikroorm.md

```diff
@@ -0,0 +1,288 @@
+
+Alternatively we can [configure the CLI](https://mikro-orm.io/docs/installation#setting-up-the-commandline-tool) by creating a configuration file `mikro-orm.config.ts` and then call the `forRoot()` without any arguments. This won't work when you use a build tools that use tree shaking. 
+
+```typescript
+@Module({
+  imports: [
+    MikroOrmModule.forRoot(),
+  ],
+  ...
+})
+export class AppModule {}
+```
+
+Afterward, the `EntityManager` will be available to inject across entire project (without importing any module elsewhere).
+
+```ts
+import { MikroORM } from '@mikro-orm/core';
+import { EntityManager } from '@mikro-orm/mysql'; // Import EntityManager from your driver package or `@mikro-orm/knex`
+
+@Injectable()
+export class MyService {
+
+    constructor(private readonly orm: MikroORM,
+                private readonly em: EntityManager) {
+    }
+
+}
+```
+
+> info **info**  Notice that the `EntityManager` is imported from the `@mikro-orm/driver` package, where driver is `mysql`, `sqlite`, `postgres` or what driver you are using.
+> 
+> In case you have `@mikro-orm/knex` installed as a dependency, you can also import the `EntityManager` from there.
+
+#### Repositories
+MikroORM supports the repository design pattern. For every entity we can create a repository. Read the complete [documentation on repositories here](https://mikro-orm.io/docs/repositories). To define which repositories shall be registered in the current scope you can use the `forFeature()` method. For example, in this way:
+
+> info **info**  You should **not** register your base entities via `forFeature()`, as there are no
+> repositories for those. On the other hand, base entities need to be part of the list
+> in `forRoot()` (or in the ORM config in general).
+
+```typescript
+// photo.module.ts
+
+@Module({
+  imports: [MikroOrmModule.forFeature([Photo])],
+  providers: [PhotoService],
+  controllers: [PhotoController],
+})
+export class PhotoModule {}
+```
+
+and import it into the root `AppModule`:
+
+```typescript
+// app.module.ts
+@Module({
+  imports: [MikroOrmModule.forRoot(...), PhotoModule],
+})
+export class AppModule {}
+```
+
+In this way we can inject the `PhotoRepository` to the `PhotoService` using the `@InjectRepository()` decorator:
+
+```typescript
+@Injectable()
+export class PhotoService {
+  constructor(
+    @InjectRepository(Photo)
+    private readonly photoRepository: EntityRepository<Photo>
+  ) {}
+
+  // ...
+}
+```
+
+#### Using custom repositories
+
+When using custom repositories, we can get around the need for `@InjectRepository()`
+decorator by naming our repositories the same way as `getRepositoryToken()` method do:
+
+```ts
+export const getRepositoryToken = <T> (entity: EntityName<T>) => `${Utils.className(entity)}Repository`;
+```
+
+In other words, as long as we name the repository same was as the entity is called,
+appending `Repository` suffix, the repository will be registered automatically in
+the Nest.js DI container.
```

Comment:
Love the mention of how this works here! Does it work this way as well for named connections, or does Mikro handle that similarly to `@nestjs/typeorm`?

Reply:
Nope, currently the adapter does not support multiple connections (MikroORM itself does, it's just a DI issue with the key names - we will need to add some `@InjectEntityManager(name?: string)` to overcome the naming conflicts, and add the `name` parameter to the `@InjectRepository` too.

There is some context in this issue: https://github.com/mikro-orm/nestjs/issues/20

I'll probably spend some time improving the adapter once v5 is getting to RC, currently this is not really a priority, but when it will be supported, the docs will be updated.